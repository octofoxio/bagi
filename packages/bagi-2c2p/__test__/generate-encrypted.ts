import * as puppeteer from 'puppeteer'
import { readFileSync } from 'fs'
import { join } from 'path'

export async function generateEncryptedCreditcard() {
  const browser = await puppeteer.launch()
  const page = await browser.newPage()
  const pageContent = readFileSync(join(__dirname, './encrypted-gen.html')).toString()
  await page.setContent(pageContent)
  await page.evaluate(() => {
    //@ts-ignore:
    My2c2p.getEncrypted('fake', function(encryptedData, errCode, errDesc) {
      console.log(encryptedData, errCode, errDesc)
      document.getElementById('result')!.innerHTML = encryptedData.encryptedCardInfo
    })
  })
  const text = await page.$eval('#result', e => e.innerHTML)
  await page.close()
  await browser.close()
  return text
}
