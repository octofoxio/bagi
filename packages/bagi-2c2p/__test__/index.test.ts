import { generateEncryptedCreditcard } from './generate-encrypted'
import { Bagi2C2PService, CurrencyCode } from '..'
import { readFileSync } from 'fs'
import { join } from 'path'

jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000
describe('2C2P service testing', () => {
  const bagi2c2p = new Bagi2C2PService('JT04', 'QnmrnH6QE23N')
  bagi2c2p.setCurrencyCode(CurrencyCode.THB)
  bagi2c2p.setCountry('TH')
  bagi2c2p.setUseSandboxAPI()
  bagi2c2p.loadConfigurationFromEnv()
  it('should create creditcard payload', async () => {
    const encryptedCreditcard = await generateEncryptedCreditcard()
    const payload = bagi2c2p.makeCreditcardS2BPaymentPayload(
      'rungsikorn rungsikavanich',
      'rungsikorn@me.com',
      'S1 best of Yui Hatano',
      1299,
      encryptedCreditcard,
      'payment-unique-id',
      'https://www.google.com'
    )
    // simulate redirect payment page
    const paymentResult = await bagi2c2p.submitPaymentPayload(payload)
    expect(paymentResult).not.toBeUndefined()
  })

  if (bagi2c2p.isPEMLoaded) {
    it('should able to decrypt response', async () => {
      const payload = readFileSync(join(__dirname, './payment-response.txt')).toString()
      const message = await bagi2c2p.decryptResponse(payload)
      expect(message).not.toBeUndefined()
      expect(message).toMatchSnapshot()
    })
  } else {
    console.warn('SKIP Decrypt testing, PEM key is not load')
    it.skip('should able to decrypt response', async () => {})
  }

  it.skip('should able to send s2s payment', async () => {
    const encryptedCreditcard = await generateEncryptedCreditcard()
    const payload = await bagi2c2p.makeCreditcardS2SPaymentPayload(
      'rungsikorn rungsikavanich',
      'S1 best of Yui Hatano',
      1299,
      encryptedCreditcard,
      'payment-unique-id'
    )
    //@ts-ignore
    const result = await bagi2c2p.submitPaymentPayload(payload)
  })
})
