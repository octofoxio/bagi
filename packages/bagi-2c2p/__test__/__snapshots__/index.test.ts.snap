// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2C2P service testing should able to decrypt response 1`] = `
Object {
  "merchantId": "764764000001696",
  "remark": "https://staging.reeeed.com",
  "transactionId": "1556952714871",
}
`;

exports[`2C2P service testing should able to send s2s payment 1`] = `"00acIO9SEX8hTkAK67FsWP4ZjQzM267cX2aKE+SD1E5lBYZDy+16JYAfO0l/XkuH5sIwGA87hlILgSppSWlj3yJbZ4IBb7UivIvnvf2uEhhLfdybB6zoc5Sykqnh5mQbEw9OhqfRSpR3Wak7r5HdhHunEqRIgwixfZwOLpfXa2rzr7Y=U2FsdGVkX18Vg33KYRXha5BysDTMnvK4ueL6yf+g+Olfby3RmugK6ZhMpe8uP0Iz"`;
